torch/distributed/rpc/backend_registry.py

```diff
@@ -21,14 +26,56 @@ def register_backend(backend_name, init_backend_handler):
             `_init_rpc()` function is called with a backend.
              This returns the agent.
     """
-    backend_registry = _get_backend_registry()
-    if backend_name in backend_registry:
+    if _is_backend_name_registered(backend_name):
         raise RuntimeError("RPC backend {}: already registered".format(backend_name))
-    backend_registry[backend_name] = init_backend_handler
+    # Create a new enum type, `BackendType`, with extended members.
+    global BackendType
+    existing_enum_dict = {member.name: member.value for member in BackendType}
+    extended_enum_dict = dict(
```

Comment:
why do we need to have two dicts here instead of directly inserting the new backend?

Reply:
Becuase Enum type is a immutable datastrucuture. There is no API from the std library to insert a new enum variant. I have todeclare a new custom Enum class dynamically.

Reply:
I was talking about have two dicts here. I see.